File: Recipes.Api/Application/Dto/RecipeDetailDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Recipes.Api.Application.Entities;

namespace Recipes.Api.Application.Dto
{
    public class RecipeDetailDto
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string[] Keywords { get; set; }
        public string Author { get; set; }
        public string ImagePath { get; set; }
        public int TimeInMin { get; set; }
        public int PersonCount { get; set; }
        public int LikesCount { get; set; }
        public int StarsCount { get; set; }
        public bool IsStarSet { get; set; }
        public bool IsLikeSet { get; set; }
        public bool IsEditable { get; set; }
        public IngredientDto[] Ingredients { get; set; }
        public StepDto[] Steps { get; set; }
    }
}

File: RecipesFrontend/src/app/dto/recipe-detail/recipe-detail-dto.ts
import { IngredientDto } from "../ingredient/ingredient-dto";
import { StepDto } from "../step/step-dto";

export interface RecipeDetailDto {
  id: number,
  title: string,
  description: string,
  keywords: string[],
  author: string,
  imagePath: string, 
  timeInMin: number,
  personCount: number,
  likesCount: number,
  starsCount: number,
  isStarSet: boolean,
  isLikeSet: boolean,
  ingredients: IngredientDto[],
  steps: StepDto[],
}
